### ISSUE TITLE
Unable to collect styled-component styleTags for SSR.

### ISSUE THREAD

**Issue by der1igedon** [grommet/grommet#6976]
Hello,

I try to collect the styleTags of a grommet.io page in order to use them for server-side-rendering with client hydration. I use ServerStyleSheet() as described here: 

https://styled-components.com/docs/advanced#server-side-rendering

The problem is, that there are no styles collected with grommet components used. If I use other components and apply styled-components to it, it works just fine. I setup a simple example:

https://github.com/der1igedon/vite-ssr-styled-components-grommet/tree/main/

The styles should be collected on the server side (see renderer/_default.page.server.tsx) and injected in the <style>-Tag of the documentHtml. Unfortunately there are no styles detected. See 
`  console.log("Found styles: ", sheet.getStyleTags());`

Can anyone tell me how this can be achieved?
Thank you!

**Comment by AurelienLourot** [comment]
TL;DR I think the problem is not in Grommet. The issue must be somewhere in https://github.com/tdonhauser/vite-ssr-styled-components-grommet

Hey @tdonhauser I have spent a few hours wrapping my head around the issue. I couldn't figure out what the problem is. Indeed `getStyleTags()` doesn't catch the grommet components' styles during the server-side rendering and we experience a FOUC. I played a lot with your code and can't figure out the culprit. I'm wondering if it has anything to do with `noExternal: ["styled-components", "@emotion/*"],` at https://github.com/tdonhauser/vite-ssr-styled-components-grommet/blob/main/vite.config.ts#L8 . But I also can't find a way to remove `noExternal` in your project.

So I decided to start from scratch, the way I would do it myself. I scaffolded a brand new [vike-react](https://github.com/vikejs/vike-react) project with [Bati](https://batijs.github.io/). Then I added styled-components and Grommet. Everything works as expected: the server-side rendered page contains all the styles, no FOUC. I didn't need `noExternal` interestingly.

Here is the result: https://github.com/vikejs/vike-react-styled-components-grommet

Check out the readme and the commit messages, in which I'm walking you through what I did, step by step. Let me know if you have any questions.

Sorry I couldn't spot the root cause of your issue. It's not very satisfying. But thanks a lot for digging into this in the first place! I hope this helps

**Comment by tdonhauser** [comment]
wow thanks a lot! that's great. I'll take a look on that 👍
